### DISCUSSION TITLE
Migrating database file paths after chaging upload folder dir

### DISCUSSION THREAD

**Discussion by p4block** [immich-app/immich#9655] [Q&A]
My images used to be in `/photos` and the upload dir was set as such. I've now moved them somewhere else, working for new uploads, but all old uploads are borked.

I've noticed that the database has full filepaths instead of relative. I can do a few SQL queries to swap them all to new path, but I wonder if any button in the Immich admin panel is supposed to do this.  I've tried running the library tasks etc but nothing so far.

I've searched for a while and I can only find that it should be working it out by itself, but it's not.

**Comment by bo0tzz** [comment]
Without more detail such as all the different paths you've moved things between etc, it's impossible to say what's going wrong.

**Reply by ChocolateLoverRaj** [reply]
What's wrong is that in the postgres db the paths are absolute, so when the files are moved to a different folder, all the paths stored in the postgres db are wrong (causing images to not load). 

For example in https://github.com/immich-app/immich/discussions/15356 I came across this problem.

I think that relative paths should be stored in the postgres db to avoid this problem.

**Comment by ChocolateLoverRaj** [comment]
As described in #15356, there is a workaround.

https://stackoverflow.com/a/49063440/11145447
